network/framework/src/application/storage.rs

```diff
@@ -220,7 +251,16 @@ impl PeersAndMetadata {
             // have multiple connections for a peer
             let active_connection_id = entry.get().connection_metadata.connection_id;
             if active_connection_id == connection_id {
-                entry.remove()
+                let peer_metadata = entry.remove();
+                let nc = self.get_network_context(&peer_network_id.network_id());
+                // TODO: get some actual DisconnectReason in here?
```

Comment:
Why the comment? The reason is passed to `remove_peer_metadata`, right?

Reply:
stale TODO is done!